tfjs-layers/src/layers/nlp/modeling/cached_multihead_attention.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+/**
+ *  Cached MHA layer based on `MultiHeadAttention`.
+ */
+
+/* Original source: keras_nlp/layers/modeling/cached_multi_head_attention.py */
+import { Tensor, Tensor1D, Tensor2D, serialization } from '@tensorflow/tfjs-core';
+
+import { MultiHeadAttention } from '../multihead_attention';
+import { NotImplementedError } from '../../../errors';
+
+export declare interface CachedMultiHeadAttentionOptions {
+  /**
+   * Query `Tensor` of shape `(B, T, dim)`.
+   */
+
```

Comment:
Thanks for the review! This was done purposely since I'm using `query` as technically the first argument in `call` (just renamed `inputs` to `query` [here](https://github.com/tensorflow/tfjs/pull/7842/files#diff-2046d47f1b658fb5f620156abb5b18f61d014ed230165badd98a2c6d959b0d31R107)), but wanted to add this spec about it somewhere. This pattern happens in other layers too. Is there another way I can include this `query` doc?

I didn't want to include `query` here and then just have an unused first parameter to `call`. Let me know what you think - thanks!